### PR TITLE
Add MarkDistinct support

### PR DESCRIPTION
**Pull request by shiyu-bytedance**
Add MarkDistinct support.

   This is the prestissimo companion to https://github.com/facebookincubator/velox/pull/4173

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 1]
```diff
@@ -1946,6 +1971,10 @@ core::PlanNodePtr VeloxQueryPlanConverterBase::toVeloxQueryPlan(
           std::dynamic_pointer_cast<const protocol::WindowNode>(node)) {
     return toVeloxQueryPlan(window, tableWriteInfo, taskId);
   }
+  if (auto markdistinct =
```
Comment (aditi-pandit) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1974]: Nit : Rename to markDistinct.

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 2]
```diff
@@ -1869,6 +1869,31 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
...
 
+std::shared_ptr<const velox::core::MarkDistinctNode> VeloxQueryPlanConverterBase::toVeloxQueryPlan(
+    const std::shared_ptr<const protocol::MarkDistinctNode>& node,
+    const std::shared_ptr<protocol::TableWriteInfo>& tableWriteInfo,
+    const protocol::TaskId& taskId) {
+  const core::FieldAccessTypedExprPtr markerVariable =
+      exprConverter_.toVeloxExpr(node->markerVariable);
+
+  std::vector<core::FieldAccessTypedExprPtr> distinctVariables;
+  for (auto dVar : node->distinctVariables) {
```
Comment (aditi-pandit) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1880]: Nit : Make this distinctVariable to make it consistent with hashVariable below.

Comment (aditi-pandit) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1880]: Make this const auto& ?

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveMarkDistinctQueries.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import org.testng.annotations.Test;
+
+public class TestHiveMarkDistinctQueries
+        extends AbstractTestHiveQueries
+{
+    private static final String DISTINCT_AGGREGATE = "WITH sample AS ( SELECT * FROM (VALUES (1,1,1), (1,1,2), (1,1,1), (2,1,2), (2,2,1), (2,3,2)) AS account (c1, c2, c3) ) SELECT c1,sum(distinct c2),sum(distinct c3) from sample group by c1";
```
Comment (aditi-pandit) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveMarkDistinctQueries.java:21]: Why is this private static final variable and can't be local to testDistinctAggregate ?

Do you anticipate many tests for MarkDistinct Operator ? Do we need a new file for this ?

**Review by aditi-pandit — COMMENTED** [review]
@shiyu-bytedance : Thanks for this PR. Have bunch of nits.

**Comment by yzhang1991** [comment]
Hi @shiyu-bytedance, can you follow up on this PR? Thanks!

**Comment by mbasmanova** [comment]
This is a dup of #19846

**Pull request closed without merging**